HighAvailabilityModule.Storage.Client/SQLStorageMembershipClient.cs

```diff
@@ -0,0 +1,392 @@
+            string type = getDataEntry.Item2;
+
+            if (type == "System.String")
+            {
+                return value.ToString();
+            }
+            else
+            {
+                throw new FormatException("Input value is not string.");
+            }
+        }
+
+        public async Task<int> TryGetInt(string path, string key)
+        {
+            var getDataEntry = await GetDataEntryAsync(path, key);
+            string value = getDataEntry.Item1;
+            string type = getDataEntry.Item2;
+
+            if (type == "System.Int32")
+            {
+                return Int32.Parse(value); 
+            }
+            else
+            {
+                throw new FormatException("Input value is not int.");
+            }
+        }
+
+        public async Task<long> TryGetLong(string path, string key)
+        {
+            var getDataEntry = await GetDataEntryAsync(path, key);
+            string value = getDataEntry.Item1;
+            string type = getDataEntry.Item2;
+
+            if (type == "System.Int64")
+            {
+                return Int64.Parse(value);
+            }
+            else
+            {
+                throw new FormatException("Input value is not long.");
+            }
+        }
+
+        public async Task<double> TryGetDouble(string path, string key)
+        {
+            var getDataEntry = await GetDataEntryAsync(path, key);
+            string value = getDataEntry.Item1;
+            string type = getDataEntry.Item2;
+
+            if (type == "System.Double")
+            {
+                return Double.Parse(value);
+            }
+            else
+            {
+                throw new FormatException("Input value is not double.");
+            }
+        }
+
+        public async Task<string[]> TryGetStringArray(string path, string key)
+        {
+            var getDataEntry = await GetDataEntryAsync(path, key);
+            string value = getDataEntry.Item1;
+            string type = getDataEntry.Item2;
+
+            if (type == "System.String[]")
+            {
+                return value.Split(",");
+            }
+            else
+            {
+                throw new FormatException("Input value is not string[].");
+            }
+        }
+
+        public async Task<byte[]> TryGetByteArray(string path, string key)
+        {
+            var getDataEntry = await GetDataEntryAsync(path, key);
+            string valueTmp = getDataEntry.Item1;
+            string type = getDataEntry.Item2;
+
+            if (type == "System.Byte[]")
+            {
+                string[] s = valueTmp.Split(",");
+                byte[] value = new byte[s.Length];
+                for (int i=0; i<s.Length; i++)
+                {
+                    value[i] = byte.Parse(s[i]);
+                }
+                return value;
+            }
+            else
+            {
+                throw new FormatException("Input value is not byte[].");
```

Comment:
Please refer to the usage of ![FormatException](https://docs.microsoft.com/en-us/dotnet/api/system.formatexception?view=netframework-4.8#remarks). Is `InvalidOperationException` properer here?